app/src/main/java/de/westnordost/streetcomplete/data/osm/mapdata/MapDataCache.kt

```diff
@@ -213,7 +222,7 @@ class MapDataCache(
     ): List<Element> = synchronized(this) {
         val cachedElements = keys.mapNotNull { key ->
             when (key.type) {
-                ElementType.NODE -> spatialCache.get(key.id)
+                ElementType.NODE -> spatialCache.get(key.id) ?: nodeCache[key.id]
```

Comment:
This construct `spatialCache.get(key.id) ?: nodeCache[key.id]` occurs all over the place, would make sense to put this into a function. Not only for convenience, but also to have a better overview whether it is done everywhere consistently (i.e. you did not miss one)

Reply:
Done. `spatialCache.get` is still called durig `trim` and `update` for handling `relationIdsByElementKeyCache` and `wayIdsByNodeIdCache`